src/interceptor/mod.rs

```diff
@@ -60,20 +60,17 @@ pub trait Interceptor: Send + Sync {
     ///
     /// The returned future is allowed to borrow the interceptor for the
     /// duration of its execution.
-    fn intercept<'a>(
-        &'a self,
-        request: Request<AsyncBody>,
-        ctx: Context<'a>,
-    ) -> InterceptorFuture<'a, Self::Err>;
+    fn intercept(&self, request: Request<AsyncBody>, ctx: Context) -> InterceptorFuture<Self::Err>;
```

Comment:
This makes the interceptor API (yet unstable though it is) a bit more clunky to implement. Since an interceptor only requires a shared reference to invoke, and since we now clone the entire client and hold onto it for the duration of a `ResponseFuture`, should it be theoretically possible to keep allowing `InterceptorFuture` to borrow the interceptor?

Reply:
As we returned `InterceptorFuture` in `send` directly, I thought we should make `InterceptorFuture` static too.

https://github.com/sagebind/isahc/blob/532237b48db554d824d3ed020782b87fa95feb54/src/interceptor/context.rs#L15-L27

Reply:
Hmm, I see. While the value of having a `ResponseFuture` free of lifetimes is apparent to me, it isn't as clear as to what the benefit would be for `InterceptorFuture`. From the perspective of a user who might implement an interceptor, I'm not sure of a use-case for having the future returned by `Context::send` be able to escape my `intercept` call; I suppose you could store open connections in a concurrent hashmap, but to what end? And needing to return my _own_ `InterceptorFuture` as well, it means extra cloning as demonstrated in the updates needed to the various builtin interceptors.

Reply:
I got it. Let me have a try. I'm not sure can we construct a static `ResponseFuture` with `InterceptorFuture` carries a lifetime.

Reply:
By returning a `InterceptorFuture` directly with `send`, probably not, but a cloned `HttpClient` plus an `&InterceptorObj` together should be possible to return in a `'static` future, as long as the `HttpClient` is moved into it.

Reply:
If it isn't possible to implement without difficulty then that's probably fine as well since the `Interceptor` interface isn't final for now anyway and might need some revision to ensure ease of use. Though of course allowing borrowing helps the user-friendliness.

Reply:
`CookieJar` is wrappered via `Arc`, so cloning it is cheap.

And I refactored default headers interprtor in https://github.com/sagebind/isahc/pull/405/commits/60965f4fc80ef025a722763672856717cfd48ffc , so that we don't need to clone the entire `HeaderMap`. (And we can remove an extra `Box::pin`!)

So far, I feel like make `InterceptorFuture` static looks good now.